### PR TITLE
Better exceptions handling, COM-106

### PR DESCRIPTION
**Pull request by michaljurecko**
Changes:
- Updated version of `league/flysystem` 1.0.56 => 1.0.64
- Updated version in `docker-compose.yml` version 2.0 -> 2.2
  - phpstorm can run phpunit in docker environment without problems
- Modified code to use new `League\Flysystem\FilesystemException`
- Added clear msg for `ftp_rawlist()...` problem, #11 
- All covered by tests

### CODE REVIEW

File: src/Exception/ExceptionConverter.php [hunk 1]
```diff
@@ -43,6 +30,20 @@ public static function handleDownloadException(\Throwable $e): void
             ));
         }
 
+        self::handleCommonException($e);
+    }
+
+    private static function handleCommonException(\Throwable $e): void
```
Comment (ondrajodas) [inline src/Exception/ExceptionConverter.php:36]: tady by si měl ještě odchytávat interface `FlysystemException `

File: src/Exception/ExceptionConverter.php [hunk 2]
```diff
@@ -51,6 +52,13 @@ public static function handleDownloadException(\Throwable $e): void
             ));
         }
 
+        // Catch user_error from phpseclib
+        // phpcs:disable
+        if (preg_match_all('/(getaddrinfo failed)|(Could not connect to)|(Cannot connect to)|(Root is invalid)|(The authenticity of)|(Connection closed prematurely)/', $e->getMessage())) {
```
Comment (ondrajodas) [inline src/Exception/ExceptionConverter.php]: tady potom bude zbytečný `Could not connect to `, `Root is invalid`

**Review by ondrajodas — CHANGES_REQUESTED** [review]
Tady by bylo fajn všechny ty tři metody nahradit jednou `handleException` a v té to zpracovávat

File: src/Exception/ExceptionConverter.php (hunk 2 above)
Comment (ondrajodas) [reply src/Exception/ExceptionConverter.php]: ty message jsou odchytnutý právě `FlysystemException` v PR
https://github.com/thephpleague/flysystem/pull/1070/files

File: src/Exception/ExceptionConverter.php (hunk 1 above)
Comment (michaljurecko) [reply src/Exception/ExceptionConverter.php:36]: - `handleCopyFilesException` a `handlePrepareToDownloadException` by som nespajal, kedze je to to tak lepsie rozsiritelne, ... do buducna moze byt nejaka Exception, ktora patri iba k jednemu z tych dvoch, ... alebo budeme potrebovat inu spravu ak sa jedna o `copy` a inu ak sa jedna o `download`, ... tak su riesene aj vsetky testy.
- interface `FlysystemException` neexistuje, ... tak bol navrhnuty PR do tej externej kniznice, ale oni si to pred vydaním este upravili ... tomu zodpoveda `League\Flysystem\FilesystemException`, ... takze je to IMHO spravne

File: src/Exception/ExceptionConverter.php (hunk 2 above)
Comment (michaljurecko) [reply src/Exception/ExceptionConverter.php]: mas pravdu, tie som mal tiez odstranit. idem to fixnut.

Comment (michaljurecko) [reply src/Exception/ExceptionConverter.php]: zmene v: https://github.com/keboola/ex-ftp/pull/36/commits/5f5ad3b12d127cc2d54692ded99605df59b976c7

File: src/Exception/ExceptionConverter.php (hunk 1 above)
Comment (michaljurecko) [reply src/Exception/ExceptionConverter.php:36]: https://github.com/thephpleague/flysystem/pull/1070#issuecomment-570794787

Comment (michaljurecko) [reply src/Exception/ExceptionConverter.php:36]: https://github.com/thephpleague/flysystem/commit/c63d001e8fbd0e795e4e84d52bb56eab0028c29a

**Review by ondrajodas — APPROVED** [review]
👍 supr

**Pull request merged**